src/attrs_strict/_type_validation.py

```diff
@@ -248,6 +258,38 @@ def _handle_union(attribute, value, expected_type):
     raise UnionError(value, attribute.name, expected_type)
 
 
+def _handle_literal(attribute, value, expected_type):
+    flattened_literals = _flatten_literals(expected_type)
+
+    if not any(
+        value == literal and type(value) == type(literal)
+        for literal in flattened_literals
+    ):
+        raise LiteralError(attribute.name, value, flattened_literals)
+
+
+def _flatten_literals(literals):  # type: ignore
+    meta = "__args__" if hasattr(literals, "__args__") else "__values__"
+    extracted_literals = getattr(literals, meta, None)
+
+    flattened_literals = []
+    unsupported_literals = []
+
+    for literal in extracted_literals:
+        base_type = _get_base_type(literal)
+        if base_type == Literal or base_type == type(Literal):  # type: ignore
+            flattened_literals.extend(_flatten_literals(literal))
+        elif literal is None or isinstance(literal, SupportedLiterals):
```

Comment:
SupportedLiterals is a Tuple but it should be a set. This way you're doing a sequential lookup rather than a hash based one.

Reply:
isinstance takes a tuple